src/content/docs/en/guides/upgrade-to/v6.mdx

```diff
@@ -310,6 +314,25 @@ export default defineConfig({
 
 <ReadMore>Learn more about [Internationalization routing](/en/guides/internationalization/#routing).</ReadMore>
 
+### Changed: endpoints with a file extension cannot be accessed with a trailing slash
+
+<SourcePR number="14457" title="feat!: trailing slash never for endpoints with file extension"/>
+
+In Astro v5.0, endpoints with a pathname ending with a file extension (eg. `/src/pages/sitemap.xml.ts`) could be accessed with a trailing slash (`/sitemap.xml/`) or without (`/sitemap.xml`), no matter the value of `build.trailingSlash`.
+
+In Astro v6.0, these endpoints can only be accessed without a trailing slash.
+
+#### What should I do?
+
+Review your links and remove any referencing an endpoint ending with a file extension and a trailing slash:
+
+```html del={1} ins={2} title="src/pages/index.astro"
+<a href="/sitemap.xml/">Sitemap</a>
+<a href="/sitemap.xml">Sitemap</a>
+```
+
+<ReadMore>Learn more about [`build.trailingSlash`](/en/reference/configuration-reference/#trailingslash).</ReadMore>
```

Comment:
```suggestion
<ReadMore>Learn more about [custom endpoints](/en/guides/endpoints/).</ReadMore>
```
I think taking people to endpoints is more helpful here, since the config actually doesn't matter for this particular thing!

Also, I think we should say something *on* the endpoint guide about what the resulting URL will look like, and have a statement that regardless of `build.trailingSlash`, these URLs will not be accessible with a trailing slash.